creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -377,37 +388,73 @@ class SnapbackSM {
    * }
    */
   async determineNewReplicaSet ({ primary, secondary1, secondary2, wallet, unhealthyReplicasSet, healthyNodes }) {
-    let currentNodeHasExistingUserState = true
+    let newReplicaNodesSet = new Set()
     let i = 0
-    let newSecondary
-    while (currentNodeHasExistingUserState && i++ < MAX_SELECT_NEW_REPLICA_SET_ATTEMPTS) {
-      // Make sure current replica set will not be chosen as potential nodes for the new replica set
-      const healthyPotentialPeers = healthyNodes.filter(node => node !== primary && node !== secondary1 && node !== secondary2)
-
+    let issueReconfig = false
+    let newPrimary, newSecondary1, newSecondary2
+    while (newReplicaNodesSet.size !== NUMBER_OF_REPLICA_SET_NODES && i++ < MAX_SELECT_NEW_REPLICA_SET_ATTEMPTS) {
       // Select a random node that is not from the current replica set
-      newSecondary = healthyPotentialPeers[Math.floor(Math.random() * healthyPotentialPeers.length)]
+      let randomHealthyNode = healthyNodes[Math.floor(Math.random() * healthyNodes.length)]
```

Comment:
nit - lodash (existing import) has a helper for this https://lodash.com/docs/4.17.15#sample